### PR TITLE
Add try_access method to GenericReceiver

### PR DESCRIPTION
**Pull request by jean-airoldie**
This method allows the caller to apply a function to the oldest value of the channel (if any) and retrieve its output. I also fixed various typos in the tests that I left.

An example use case for this addition would be to decide whether to dequeue the next element of the channel based on its value (e.g. size).

Note that this is a breaking change since the required method `Ringbuf::peek` was added.

### CODE REVIEW

**Comment by jean-airoldie** [comment]
Also this method could be potentially misused since the subsequent try_receive call is not guaranteed to return the same value in the case of multiple consumers. So this really only makes sense to use in a single consumer setting.

File: src/buffer/ring_buffer.rs [hunk 1]
```diff
@@ -240,6 +255,12 @@ mod if_alloc {
             assert!(self.buffer.len() > 0);
             self.buffer.pop_front().unwrap()
         }
+
+        #[inline]
+        fn peek(&self) -> &Self::Item {
+            assert!(self.buffer.len() > 0);
```
Comment (Matthias247) [inline src/buffer/ring_buffer.rs:261]: I think this is not necessary given that `unwrap` is used in the next line. If the intention is to add more context information `unwrap` could be replaced with `.expect`

**Review by Matthias247 — COMMENTED** [review]
Hmm, one thing I'm not sure about is the problem you mentioned: If you use this in with `try_access` and then `try_receive` sequences then there are no guarantees that the objects worked on are the same.

One other approach I could think of is passing a callback function with a return value, which would indicate whether the value should be dropped. But then the callback also couldn't take a reference to them item because it then couldn't take it out - it would require rather an `&mut Option<T>` so that the caller could take it out. But on the other hand that's also kind of confusing to use.

**Comment by jean-airoldie** [comment]
I agree that the API is kinda confusing, also breaking change just for that.

I ended up not using this method anywhere in my codebase so feel free to close this PR. Can't remember exactly what i used instead.

**Comment by Matthias247** [comment]
Ok thanks for the comment and the previous work!

I will close it - and if someone finds a need for it and can come up with a better API we can revive it.

**Pull request closed without merging**